predicates-push-down.md

```diff
@@ -0,0 +1,151 @@
+| id                      | estRows | task      | access object | operator info                          |
++-------------------------+---------+-----------+---------------+----------------------------------------+
+| Selection_7             | 2.00    | root      |               | eq(substring("123", test.t.a, 1), "1") |
+| └─TableReader_6         | 2.00    | root      |               | data:TableFullScan_5                   |
+|   └─TableFullScan_5     | 2.00    | cop[tikv] | table:t       | keep order:false, stats:pseudo         |
++-------------------------+---------+-----------+---------------+----------------------------------------+
+```
+
+In this query, there is a predicate `substring('123', a, 1) = '1'`.
+
+From the `explain` results, we can see that the predicate is not pushed down to TiKV for calculation. This is because the TiKV coprocessor does not support the built-in function `substring`.
+
+### Case 5: predicates of inner tables on the outer join can't be pushed down
+
+```sql
+create table t(id int primary key, a int not null);
+create table s(id int primary key, a int not null);
+explain select * from t left join s on t.a = s.a where s.a is null;
++-------------------------------+----------+-----------+---------------+-------------------------------------------------+
+| id                            | estRows  | task      | access object | operator info                                   |
++-------------------------------+----------+-----------+---------------+-------------------------------------------------+
+| Selection_7                   | 10000.00 | root      |               | isnull(test.s.a)                                |
+| └─HashJoin_8                  | 12500.00 | root      |               | left outer join, equal:[eq(test.t.a, test.s.a)] |
+|   ├─TableReader_13(Build)     | 10000.00 | root      |               | data:TableFullScan_12                           |
+|   │ └─TableFullScan_12        | 10000.00 | cop[tikv] | table:s       | keep order:false, stats:pseudo                  |
+|   └─TableReader_11(Probe)     | 10000.00 | root      |               | data:TableFullScan_10                           |
+|     └─TableFullScan_10        | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo                  |
++-------------------------------+----------+-----------+---------------+-------------------------------------------------+
+6 rows in set (0.00 sec)
+```
+
+In this query，there is a predicate `s.a is null` on the inner table `s`。
+
+From the `explain` results，we can see that the predicate is not pushed below join operator. This is because the outer join will fill the inner table with `NULL` values when the on condition isn't satisfied, and the predicate `s.a is null` is used to filter the results after the join. If it is pushed down to the inner table below join, the execution plan is not equivalent with the original one.
```

Comment:
```suggestion
From the `explain` results，we can see that the predicate is not pushed below join operator. This is because the outer join fills the inner table with `NULL` values when the `on` condition isn't satisfied, and the predicate `s.a is null` is used to filter the results after the join. If it is pushed down to the inner table below join, the execution plan is not equivalent to the original one.
```